feature_engine/timeseries/forecasting/time_lag.py

```diff
@@ -0,0 +1,174 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from feature_engine.dataframe_checks import (
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import (
+    _find_or_check_numerical_variables,
+    _check_input_parameter_variables,
+)
+
+
+class TimeSeriesLagTransformer(BaseEstimator, TransformerMixin):
+    """
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of numerical variables to transform. If None, the transformer
+        will automatically select all numerical variables.
+
+    periods: int, default=1
+        Number of periods to shift. Can be positive or negative.
```

Comment:
I cannot think of a valid use case where you would use a negative lag so I agree with your first point. 

fill-value fills only the NaN values introduced at the start of the series, in particular when you use the `period` argument and do not set the `freq` parameter. The current implementation you have here allows you to shift multiple columns at the same time, in that case you probably wouldn't want to impute the missing values introduced at the start of the time series in different columns with the same value. But the current Pandas API does allow the user to do this. If you wanted to lag one column and also fill the missing values at the beginning then it could be convenient to expose `fill-value`. In practice you'd likely use the `fillna` method of a dataframe use backfilling.

Reply:
@Morgan-Sell 

to test the variables param, we have now a generic test that we can apply to all transformers. The test is [here](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/estimator_checks.py#L102).

We also have [this other test](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/estimator_checks.py#L31) that runs the above and other multiple tests that we need to run on all our transformers.

And here you have [an example](https://github.com/feature-engine/feature_engine/blob/main/tests/test_selection/test_check_estimator_selectors.py#L46) on how to run those tests for all selector transformers. For this module, at the moment we have only 1 class, but we will have more. We should do the same.

In short, we need a script like the test_check_estimator_selectors for the forecasting module, that runs the check_estimator test from sklearn (first test in the script) and the check_estimator from feature engine (second test in this script).

If you are feeling dareful, we could also create a check for the param drop_original in the estimator_checks and run it from the master check scropt. Because other transformers also have this parameter, and would benefit from this test.

Reply:
All done ;)